tests/PropertySpec/Exec.hs

```diff
@@ -0,0 +1,69 @@
+module PropertySpec.Exec
+  ( additionSpec
+  , subtractionSpec
+  , multiplicationSpec
+  , divisionSpec
+  )where
+
+import           Data.Foldable                               (fold, foldl1)
+import           Data.List.NonEmpty                          (intersperse)
+import           ExecSpec.Util                               (num)
+import           Graphics.Implicit.ExtOpenScad.Definitions   (OVal (ONum))
+import           Graphics.Implicit.ExtOpenScad.Eval.Constant (runExpr)
+import           HaskellWorks.Hspec.Hedgehog                 (requireProperty)
+import           Hedgehog                                    (diff, forAll)
+import qualified Hedgehog.Gen                                as Gen
+import qualified Hedgehog.Range                              as Range
+import           Prelude                                     (Bool (False), Floating, String, Double, Show,
+                                                              Eq, Ord, fail, show,
+                                                              ($), (&&), (+),
+                                                              (.), (<$>), (<=), (*), (/),
+                                                              (<>), (>=), (-))
+import           Test.Hspec                                  (Spec, it)
+
+approx :: (Floating a, Ord a) => a -> a -> a -> Bool
+approx z a b = a + z >= b && a <= b + z
+
+data Op = Add | Sub | Mul | Div
+  deriving Eq
+
+instance Show Op where
+  show Add = "+"
+  show Sub = "-"
+  show Mul = "*"
+  show Div = "/"
+
+opName :: Op -> String 
+opName Add = "addition"
+opName Sub = "subttraction"
+opName Mul = "multiplication"
+opName Div = "division"
+
+fromOp :: Op -> Double -> Double -> Double
+fromOp Add = (+)
+fromOp Sub = (-)
+fromOp Mul = (*)
+fromOp Div = (/)
```

Comment:
Sum type to tie all of the names and functions together, so things aren't forgotten.